### PR TITLE
Code-related prompts

### PR DESCRIPTION
**Pull request by shakedbr**
Hi,
Following #686 I contributed the following prompts:

**Dataset**: great_code
**Prompts**:

1. _bug detection_: Given a function, predict whether there is a bug in the code.
2. _fix buggy line_: Given a function and a buggy line, fix the buggy line.
3. _function name generation_: Given a function body, generate the function name.
4. _identifier prediction no choices_: Given a code with a masked identifier, generate it, (without choices).
5. _identifier prediction with choices_: Given a code with a masked identifier, generate it, (with choices).

**Dataset**: openai_humaneval
**Prompts**:
1. _function body generation_: Given a function signature and docstring, generate the function body.
2. _test_x return value generation_ : Given a function body and a call, predict the return value of test x.

**Dataset**: code_x_glue_tc_text_to_code
**Prompts**:
1. _generate class member function given class environment_: : Given a class members (methods and fields) and a method description, generate the method.

**Dataset**: neural_code_search
**Prompts**:
1. _generate a description given code_: Given a code snippet, generate a natural language description.
2.  _generate code given a description_: Given a description, generate code.

### CODE REVIEW

**Comment by VictorSanh** [comment]
thank you for your patience @shakedbr 
These are very complicated datasets, thanks for working on this!
I left a few comments, questions and suggested fixes for missing metrics.

**Comment by shakedbr** [comment]
Hi @VictorSanh, I can't see any comments

**Review by VictorSanh — CHANGES_REQUESTED** [review]
oooops @shakedbr, my bad, I never pushed out the review I started...

File: promptsource/templates/neural_code_search/evaluation_dataset/templates.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
+dataset: neural_code_search
```
[the remark this answers is not in the document]
Comment (shakedbr) [reply promptsource/templates/neural_code_search/evaluation_dataset/templates.yaml:1]: The original task is to search for code given some natural language description (e.g. in some database)., therefore none of the above is the original task. "generate code given a description" is the closest one.

File: promptsource/templates/great_code/templates.yaml [hunk 2]
```diff
@@ -0,0 +1,133 @@
+dataset: great_code
```
[the remark this answers is not in the document]
Comment (shakedbr) [reply promptsource/templates/great_code/templates.yaml:1]: The original task - VarMisuse, aims to look for a certain type of bug - a wrong usage of some variable e.g. 
```python
a = 1
b = 2
sum_a_b = add(a, a)
```
 which can be compiled without raising any errors, but it should be 
```python
a = 1
b = 2
sum_a_b = add(a, b)
```
Thus, none of the above prompts is the **exact** original task (which requires locating and fixing the bug).

File: promptsource/templates/openai_humaneval/templates.yaml [hunk 3]
```diff
@@ -0,0 +1,188 @@
+dataset: openai_humaneval
```
[the remark this answers is not in the document]
Comment (shakedbr) [reply promptsource/templates/openai_humaneval/templates.yaml:1]: fixed

File: promptsource/templates/great_code/templates.yaml [hunk 4]
```diff
@@ -0,0 +1,133 @@
...
+      \ %}\n               {% set ns.result = ns.result + [indent * ns.indent_size]\
+      \ %}\n          {% endif %}\n          {% set ns.result = ns.result + [token\
+      \ | replace('\\\\n', '\\n'), \" \"] %}\n    {% endif %}\n{% endfor %}\n{{ns.result\
+      \ | join(\"\")}}\n\nIs there a bug in the code above?\n|||\n{{ {True: \"Yes\"\
+      , False: \"No\"}[has_bug] }}\n"
+    metadata: !TemplateMetadata
+      choices_in_prompt: true
+      metrics: []
+      original_task: false
+    name: bug detection
```
[the remark this answers is not in the document]
Comment (shakedbr) [reply promptsource/templates/great_code/templates.yaml:108]: fixed

**Comment by shakedbr** [comment]
@VictorSanh I addressed all your questions and comments

**Comment by VictorSanh** [comment]
thanks @shakedbr !

I added a few more fixes (in particular setting the variables outside of the conditional statements to avoid the `Undefined variable` error)

Could you pull from master so that you can run the tests? 
I ran them locally and all except one fail:
The last bug I have is for `fix buggy line` on `great_code` on test set (3rd example i.e. `id=2`)
Capture d’écran 2022-02-04 à 2 05 09 PM

**Comment by shakedbr** [comment]
@VictorSanh sorry for the errors.
I fixed the stuff and run the test locally, and it seems to work.

**Comment by VictorSanh** [comment]
Ran the tests locally and it works just fine!
merging now, thanks @shakedbr !

File: promptsource/templates/openai_humaneval/templates.yaml [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+templates:
+  4a108b1c-7514-488f-99ed-3ca5da70e103: !Template
+    answer_choices: null
+    id: 4a108b1c-7514-488f-99ed-3ca5da70e103
+    jinja: '{{ prompt }}
+
+      Given the following docstring, what is the function body?
```
Comment (Muennighoff) [inline promptsource/templates/openai_humaneval/templates.yaml:8]: This should probably be "GIven the above"; Anyways this prompt is bad imo as there's a function name + a docstring above.

Would be better to have e.g. "Complete the following function\n{{ prompt }}" imo

**Pull request merged**